packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2904,6 +2923,15 @@ class InputDecoration {
   /// If [InputDecorationTheme.hintFadeDuration] is null defaults to 20ms.
   final Duration? hintFadeDuration;
 
+  /// Whether the input field's height should always be greater than or equal to
+  /// the height of the [hintText], even if the [hintText] is not visible.
+  ///
+  /// The [InputDecorator] widget ignores [hintText] during layout when
+  /// it's not visible, if this flag is set to false.
+  ///
+  /// Defaults to true.
+  final bool maintainHintHeight;
```

Comment:
Why do we need this new property? I think PR is a bug fix, not a new feature, so we should just make the behavior the same as `maintainHintHeight = false` without allowing `maintainHintHeight = true`.

Reply:
This is a new (opt-in) feature. Changing the current behavior will probably be breaking (although from a UX perspective the hint text should typically be as brief as possible so it shouldn't be multiline in most cases).

Reply:
But as stated in https://github.com/flutter/flutter/issues/153183 this was (probably) broken recently, which in my opinion should be considered a regression, and should be fixed without the concern of breaking projects that rely on it.

Reply:
I don't think that's true, there's [an older issue](https://github.com/flutter/flutter/issues/52008) from 2020. I think the behavior just became more noticeable after a different bug is fixed (https://github.com/flutter/flutter/pull/148423). The input decorator widget has always been taking the hint height into account afaik.

Reply:
Ok that makes sense. Thanks for pointing out.